object/object.go

```diff
@@ -17,37 +17,35 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/client/apiutil"
 )
 
-func ToString(o client.Object) string {
-	if o.GetNamespace() == "" {
-		return fmt.Sprintf("%s/%s", reflect.TypeOf(o).Elem(), o.GetName())
+func GKNN(o client.Object) string {
+	gvk := GVK(o)
+	gknn := fmt.Sprintf("%s.%s", gvk.Kind, gvk.Group)
+	if o.GetNamespace() != "" {
+		gknn += "/" + o.GetNamespace()
 	}
-	return fmt.Sprintf("%s/%s/%s", reflect.TypeOf(o).Elem(), o.GetNamespace(), o.GetName())
+	gknn += "/" + o.GetName()
+	return gknn
 }
 
 func GVK(o client.Object) schema.GroupVersionKind {
 	return lo.Must(apiutil.GVKForObject(o, scheme.Scheme))
 }
 
-func OwnedBy(owner client.Object, mergeFrom ...metav1.ObjectMeta) metav1.ObjectMeta {
+func WithOwner[T client.Object](owner client.Object, object T) T {
+	owned := reflect.New(reflect.TypeOf(object).Elem()).Interface().(T)
 	gvk := GVK(owner)
-	object := metav1.ObjectMeta{
-		Name:        owner.GetName(),
-		Namespace:   owner.GetNamespace(),
-		Annotations: owner.GetAnnotations(),
-		Labels:      owner.GetLabels(),
-		OwnerReferences: []metav1.OwnerReference{{
-			APIVersion:         gvk.GroupVersion().String(),
-			Kind:               gvk.Kind,
-			Name:               owner.GetName(),
-			UID:                owner.GetUID(),
-			Controller:         lo.ToPtr(true),
-			BlockOwnerDeletion: lo.ToPtr(true),
-		}},
-	}
-	for _, meta := range mergeFrom {
-		lo.Must0(mergo.Merge(&object, &meta, mergo.WithOverride))
-	}
-	return object
+	owned.SetName(owner.GetName())
```

Comment:
Why are we setting the name and namespace of the thing that we are owning from the owner? Maybe I'm missing what this is intending to do but I assumed that this would just update ownerReferences. 

Maybe a comment above the function could make it more clear what this library function is intending

Reply:
Thanks for questioning this, I decided to just go w/ controller-runtime's helper in all cases, even if it isn't inlinable. Better than a weak opinion on top of it.